bson/bsoncodec/time_codec.go

```diff
@@ -29,7 +29,7 @@ type TimeCodec struct {
 var (
 	defaultTimeCodec = NewTimeCodec()
 
-	_ ValueCodec  = defaultTimeCodec
+	_ valueCodec  = defaultTimeCodec
```

Comment:
Could we add a comment describing the compile check?

Reply:
This compile-time assertion is actually redundant because the codec is already passed into `RegisterTypeEncoder` and `RegisterTypeDecoder`, so I will remove it. However, I will add a comment to the `typeDecoder` assertion below.